docs/Bigtable-batchsource.md

```diff
@@ -0,0 +1,48 @@
+# Google Cloud Bigtable Source
+
+Description
+-----------
+This source reads data from Google Cloud Bigtable.
+Cloud Bigtable is Google's NoSQL Big Data database service. 
+It's the same database that powers many core Google services, including Search, Analytics, Maps, and Gmail.
+
+Credentials
+-----------
+If the plugin is run on a Google Cloud Dataproc cluster, the service account key does not need to be
+provided and can be set to 'auto-detect'.
+Credentials will be automatically read from the cluster environment.
+
+If the plugin is not run on a Dataproc cluster, the path to a service account key must be provided.
+The service account key can be found on the Dashboard in the Cloud Platform Console.
+Make sure the account key has permission to access Bigtable.
+The service account key file needs to be available on every node in your cluster and
+must be readable by all users running the job.
+
+Properties
+----------
+**Reference Name:** Name used to uniquely identify this source for lineage, annotating metadata, etc.
+
+**Project ID**: Google Cloud Project ID, which uniquely identifies a project.
+It can be found on the Dashboard in the Google Cloud Platform Console.
+
+**Instance ID**: Google Cloud Bigtable instance ID.
+
+**Table**: Database table name.
+
+**Key Alias**: Name of the field for row key.
+
+**Scan Row Start**: Scan start row.
+
+**Scan Row Stop**: Scan stop row.
+
+**Scan Time Range Start**: The starting timestamp used to filter columns.
+
+**Scan Time Range Stop**: The ending timestamp used to filter columns.
+
+**Service Account File Path**: Path on the local file system of the service account key used for
+authorization. Can be set to 'auto-detect' when running on a Dataproc cluster.
+When running on other clusters, the file must be present on every node in the cluster.
+
+**Schema**: Specifies the schema that has to be output. 
+Only columns defined in schema will be included into output record. Field name should be in form 
+"<family>__<column>".
```

Comment:
I checked with UI and the restriction on ':' is due to an avro schema library. Avro doesn't allow ':' in field names (https://avro.apache.org/docs/1.8.2/spec.html#names). I don't think that restriction is going away any time soon.

I don't imagine most people have double underscores in the family so this is probably ok for most use cases, though ugly. This also means the plugin cannot handle other characters in the family or qualifier either though...

Perhaps we could support a 'Column Mapping' advanced property, where a user can specify that 'family:qualifier' maps to output field 'xyz'. If the user has unsupported characters in their qualifier or family they can use that mapping property. This can be done in a follow up PR to make it easier to review.

Reply:
left without changes for now